### PR TITLE
homework 1

### PR DESCRIPTION
**Pull request by CherednichenkoNikita**

### CODE REVIEW

**Review by ochedenis — DISMISSED** [review]

File: src/main/java/com/java_professional/cherednichenko/homework/lesson_1/Product.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
+package com.java_professional.cherednichenko.homework.lesson_1;
+
+import lombok.*;
```
Comment (yuriy-tsarenko) [inline src/main/java/com/java_professional/cherednichenko/homework/lesson_1/Product.java]: replace this import with single class import

File: src/main/java/com/java_professional/cherednichenko/homework/lesson_1/Product.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+import java.util.*;
```
Comment (yuriy-tsarenko) [inline src/main/java/com/java_professional/cherednichenko/homework/lesson_1/Product.java]: replace this import with single class import

File: src/main/java/com/java_professional/cherednichenko/homework/lesson_1/Product.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+@NoArgsConstructor
+@AllArgsConstructor
+@Getter
+@Setter
+@ToString
+public class Product {
```
Comment (yuriy-tsarenko) [inline src/main/java/com/java_professional/cherednichenko/homework/lesson_1/Product.java:12]: Add the description of this task.

File: src/main/java/com/java_professional/cherednichenko/homework/lesson_1/Product.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+            return Integer.compare(o1.getPrice(), o2.getPrice());
+        }
+    }
+
+    static class ProductQuantityComparator implements Comparator<Product> {
+        @Override
+        public int compare(Product o1, Product o2) {
+            return Integer.compare(o1.getQuantity(), o2.getQuantity());
+        }
+    }
```
Comment (yuriy-tsarenko) [inline src/main/java/com/java_professional/cherednichenko/homework/lesson_1/Product.java]: Merge this methods into one

**Review by yuriy-tsarenko — CHANGES_REQUESTED** [review]
Fix bugs according to my comments.
By the way, you have incorrect homework naming! Yo can read naming rules in slack.

File: java_professional.iml [hunk 5]
```diff
@@ -1,2 +1,17 @@
...
+      <sourceFolder url="file://$MODULE_DIR$/src/main/resources" type="java-resource" />
+      <excludeFolder url="file://$MODULE_DIR$/target" />
+    </content>
+    <orderEntry type="inheritedJdk" />
+    <orderEntry type="sourceFolder" forTests="false" />
+    <orderEntry type="library" name="Maven: org.apache.logging.log4j:log4j-api:2.13.2" level="project" />
+    <orderEntry type="library" name="Maven: log4j:log4j:1.2.17" level="project" />
+    <orderEntry type="library" scope="PROVIDED" name="Maven: org.projectlombok:lombok:1.18.16" level="project" />
+  </component>
+</module>
```
Comment (yuriy-tsarenko) [inline java_professional.iml]: Add this file to gitignore and remove from PR

**Review by yuriy-tsarenko — CHANGES_REQUESTED** [review]
Fix gitignore and commit changes

**Comment by yuriy-tsarenko** [comment]
fix build issues

**Pull request merged**